asyncthreadpool.nim

```diff
@@ -135,7 +140,10 @@ proc sendErrorBack[T](e: ref Exception, notifPipeW: PipeFd, c: ChannelFromPtr, f
   var msg: MsgFrom
   msg.writeResult = proc() =
     let fut = cast[Future[T]](fut)
-    fut.fail(e)
+    when not defined(ChronosAsync):
+      fut.fail(e)
+    else:
+      fut.fail(cast[ref CatchableError](e))
```

Comment:
This looks hacky. Maybe define a type alias. Like:
```nim
when defined(ChronosAsync):
  type BaseExceptionType = CatchableError
else:
  type BaseExceptionType = Exception
```
Then use `BaseExceptionType` across this file.